### PR TITLE
[MPLUGIN-433] Allow to disable link validation

### PR DESCRIPTION
**Pull request by kwin**
Also validate internal links to javadocs in mojo/parameter description and deprecated

### CODE REVIEW

File: maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/report_old/PluginReport.java [hunk 1]
```diff
@@ -218,6 +218,19 @@
...
+     * (e.g. when using an aggregator javadoc report) link validation needs to be disabled by setting
+     * this value to {@code true}.
+     * This might have the drawback that some links being generated in the report might be broken
+     * in case not all parameter types and javadoc link references are resolvable through the sites being given to
+     * {@link DescriptorGeneratorMojo}.
+     * 
+     * @since 3.7.0
+     */
+    @Parameter
+    private boolean disableInternalJavadocLinkValidation;
```
Comment (slawekjaranowski) [inline maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/report_old/PluginReport.java:232]: I would like not to add new parameters to deprecated report - what do you think?

Comment (kwin) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/report_old/PluginReport.java:232]: A lot of parameters have been added in 3.7.0 (currently both in deprecated and new report goal) not only this one. I don't think that this does any harm here.

Comment (slawekjaranowski) [reply maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/report_old/PluginReport.java:232]: ok, no problem here

**Comment by slawekjaranowski** [comment]
Please also rebase with current codebase ... build should will be ok 😄

**Comment by kwin** [comment]
> Please also rebase with current codebase

Done.

**Comment by slawekjaranowski** [comment]
What do you think about adding IT test for such options?

By the way we don't have any IT tests for javadoc links ... or I don't see.

**Comment by kwin** [comment]
> By the way we don't have any IT tests for javadoc links ... or I don't see.

Not end2end but there is a lot of unit tests and https://github.com/apache/maven-plugin-tools/blob/master/maven-plugin-tools-api/src/test/java/org/apache/maven/tools/plugin/javadoc/JavadocSiteIT.java which checks that the links being created actually work against a Javadoc site created from JDK8/11/17.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by kwin** [comment]
I now also added a proper IT in https://github.com/apache/maven-plugin-tools/pull/162/commits/6b564c8aaf89af4776ffb7898e2a86a4ce400a4f with some basic checks on generated links.

**Comment by jira-importer** [comment]
Resolve #731

**Pull request merged**